app/Jobs/Speedtests/ExecuteOoklaSpeedtest.php

```diff
@@ -61,14 +61,35 @@ public function handle(): void
         } catch (ProcessFailedException $exception) {
             $messages = explode(PHP_EOL, $exception->getMessage());
 
-            $message = collect(array_filter($messages, 'json_validate'))->last();
-
-            $this->result->update([
-                'server_id' => $this->serverId,
-                'data' => json_decode($message, true),
-                'status' => ResultStatus::Failed,
-            ]);
-
+            // Extract only the "message" part from each JSON error message
+            $errorMessages = array_map(function ($message) {
+                $decoded = json_decode($message, true);
+                if (json_last_error() === JSON_ERROR_NONE && isset($decoded['message'])) {
+                    return $decoded['message'];
+                }
+
+                return ''; // If it's not valid JSON or doesn't contain "message", return an empty string
+            }, $messages);
+
+            // Filter out empty messages and concatenate
+            $errorMessage = implode(' | ', array_filter($errorMessages));
+
+            // Add server ID to the error message if it exists
+            if ($this->serverId !== null) {
```

Comment:
Question on this, if no `serverId` is supplied (i.e. when the CLI chooses a server vs. a user supplied one) wouldn't this fail to update the result with a failured record?

Reply:
Nice catch, this was more focused on when there is a Server ID supplied that does not exist. But of course it will fail as well without a server id when there is no internet connection. 

Changed it to make the server id optional.